### ISSUE TITLE
globus-gssapi-gsi and TLS 1.3

### ISSUE THREAD

**Issue by ellert** [gridcf/gct#43]
The upcoming openssl 1.1.1 adds support for TLS 1.3. However globus-gssapi-gsi is incompatible with TLS 1.3, as can be seen from the results of "make check".

See globus/globus-toolkit#123 for details.

Openssl 1.1.1 is available in Debian experimental.

Any TLS experts available to investigate?

**Comment by fscheiner** [comment]
The following is a list of files in our repo at 4361e1e307eeb9ee73c7523d8bea19aafa674a45 that seem to include OpenSSL headers:
```
gct$ grep -nr '#include <openssl/' * | cut -d ':' -f1 | sort -u
gass/copy/source/globus_gass_copy.c
gass/copy/source/globus_gass_copy.h
gass/copy/source/globus_url_copy.c
gram/jobmanager/source/globus_gram_job_manager_gsi.c
gridftp/hdfs/src/gridftp_hdfs_cksm.c
gridftp/hdfs/src/gridftp_hdfs.h
gridftp/server/src/globus_i_gfs_data.c
gridftp/server/src/modules/file/globus_gridftp_server_file.c
gsi/gridmap_eppn_callout/globus_gridmap_eppn.c
gsi/gridmap_verify_myproxy_callout/globus_gridmap_verify_myproxy.c
gsi/gssapi/source/configure.ac
gsi/gssapi/source/test/gssapi_inquire_sec_ctx_by_oid_test.c
gsi/gssapi/source/test/mech_compatibility_test.c
gsi/proxy/proxy_ssl/source/library/proxycertinfo.c
gsi/proxy/proxy_ssl/source/library/proxycertinfo.h
gsi/proxy/proxy_ssl/source/library/proxypolicy.c
gsi/proxy/proxy_ssl/source/library/proxypolicy.h
gsi/proxy/proxy_ssl/source/test/test_proxycertinfo.c
gsi/proxy/proxy_utils/source/programs/grid-cert-diagnostics.c
myproxy/source/certauth_extensions.c
myproxy/source/myproxy_common.h
myproxy/source/myproxy_ocsp_aia.c
myproxy/source/myproxy_ocsp_aia.h
myproxy/source/myproxy_ocsp.c
myproxy/source/myproxy_ocsp.h
myproxy/source/ssl_utils.h
myproxy/source/voms_utils.h
packaging/debian/gsi-openssh/debian/patches/hpn-14.13-isshd.v3.19.1.patch
packaging/debian/gsi-openssh/debian/patches/hpn_isshd-gsi.7.5p1b.patch
packaging/debian/gsi-openssh/debian/patches/hpn_isshd-gsi_ossl.7.5p1b.patch
packaging/debian/gsi-openssh/debian/patches/openssh-7_5_P1-hpn-14.13.diff
```
...and might need patching to support TLS1.3. Not that many I think, although that doesn't say something about the amount of code that needs to be patched.

Or are changes only needed for the files in `gsi/gssapi`?

**Comment by fscheiner** [comment]
Referencing https://github.com/ellert/gct/tree/tls-v1.3 here.

**Comment by ellert** [comment]
Pull request enabling TLS v1.3 here: #150

**Comment by fscheiner** [comment]
@ellert:
> Pull request enabling TLS v1.3 here: #150

**F A N T A S T I C !** :-)

**Comment by fscheiner** [comment]
Fixed in [GCT 6.2.20210826 maintenance release](https://github.com/gridcf/gct/releases/tag/v6.2.20210826).
